### ISSUE TITLE
Set pipeline concurrency limit based on kwarg value

### ISSUE THREAD

**Issue by alfister** [mage-ai/mage-ai#5330]
### Is your feature request related to a problem? Please describe.

I'd like to have more fine-tuned control over the concurrency of pipeline runs.

For instance, if my pipeline has a `user` kwarg, I'd like to be able to limit the concurrency to 1 for each user while still allowing the pipeline to run for others in parallel.

Up to this point, I've been duplicating my pipelines for each possible kwarg value and setting pipeline concurrency to 1 for each copy. However, this is becoming unmanageable and difficult to maintain as the number of possible values grows. 

### Describe the solution you'd like

Settings to configure pipeline concurrency based on kwarg value.

This may look something like

```
concurrency_limits:
    - key: "color"
      value: "green"
      limit: 1
    - key: "color"
      value: "blue"
      limit: 2
```

### Describe alternatives you've considered

It would also be nice to have the ability to apply a limit based on uniqueness (ex. set a limit of 1 per unique value) without having to explicitly specify each unique option.

### Additional context

N/A
